core/lib/dal/src/models/storage_btc_block.rs

```diff
@@ -1,28 +1,29 @@
 use std::str::FromStr;
 
-use anyhow::Context;
-use bitcoin::Txid;
-use zksync_types::btc_block::ViaBtcBlockDetails;
+use bitcoin::{hashes::Hash, Txid};
+use zksync_types::{btc_block::ViaBtcL1BlockDetails, L1BatchNumber, H256};
 
 #[derive(Debug, Clone, sqlx::FromRow)]
-pub(crate) struct ViaBtcStorageBlockDetails {
+pub struct ViaBtcStorageL1BlockDetails {
     pub number: i64,
+    pub timestamp: i64,
     pub hash: Option<Vec<u8>>,
-    pub commit_tx_id: String,
-    pub reveal_tx_id: String,
+    pub commit_tx_id: Option<String>,
+    pub reveal_tx_id: Option<String>,
+    pub blob_id: String,
 }
 
-impl From<ViaBtcStorageBlockDetails> for ViaBtcBlockDetails {
-    fn from(details: ViaBtcStorageBlockDetails) -> Self {
-        ViaBtcBlockDetails {
-            number: details.number,
-            hash: details.hash,
-            commit_tx_id: Txid::from_str(&details.commit_tx_id)
-                .context("Failed to parse txid")
-                .unwrap(),
-            reveal_tx_id: Txid::from_str(&details.reveal_tx_id)
-                .context("Failed to parse txid")
-                .unwrap(),
+impl From<ViaBtcStorageL1BlockDetails> for ViaBtcL1BlockDetails {
+    fn from(details: ViaBtcStorageL1BlockDetails) -> Self {
+        ViaBtcL1BlockDetails {
+            number: L1BatchNumber::from(details.number as u32),
+            timestamp: details.timestamp,
+            hash: H256::from_slice(&details.hash.expect("Parse block hash")),
```

Comment:
Yes, I also noticed that the `hash` is an Option. The reason is because the SQL file `l1_block` table did not set this column as mondatory to set when create a row [here](https://github.com/vianetwork/via-core/blob/591da4a025e9389fe89fca6ab21b6b1a2ca35df2/core/lib/dal/migrations/20211026134308_init.up.sql#L54)

To remove this Option we have to set the column as NOT NULL, but I'm not sure if this will not creak the code some where.

Reply:
fixed! I changed the hash data type to `Option<vec<u8>>` to avoid cast when implement `From` trait. Now, we cast to H256 when create the inscription message [here](https://github.com/vianetwork/via-core/blob/6588bef5b4e04b0d3471730331cf03cfc07ae015/core/node/via_btc_sender/src/aggregator.rs#L131), I'm converting the Option to Result.